sdk/src/main/java/com/kin/ecosystem/main/presenter/EcosystemPresenter.java

```diff
@@ -85,6 +85,8 @@ public void onSaveInstanceState(Bundle outState) {
 	public void onDetach() {
 		super.onDetach();
 		removeBalanceObserver();
+		navigator = null;
```

Comment:
navigator is provided in the constructor and not in onAttached. can EcosystemPresenter be reused after onDetach.

protect against NPE in case related methods are called from async operations

Reply:
`ecosystemPresenter.onDetach()` is being called from view `onDestroy` so it can't be reused.